WordPress/gutenberg
# Core Data: Rename `types` directory to `entity-types`

Part of #39211 

## Description

When we introduced the types for the core entities in #38666 we used a
directory named `types`. Unfortunately this meant that we can't create the
customary file `src/types.ts` because of the name clash in module resolution.

In this patch we're renaming the directory so that we can separately
import `entity-types(/index.ts)` and also `types(.ts)` without the
naming conflict.

## Testing Instructions

Audit the types. As a type-only change this won't impact the built files.
Furthermore these types have been exported by `@wordpress/core-data`; that
shouldn't change (verify please 😄 )

## Types of changes
This is a type-only change and won't impact the build.

## Checklist:
